src/components/Admin/ManageSensors/DeleteSensorModal.tsx

```diff
@@ -0,0 +1,208 @@
+import {firestore} from '../../../firebase';
+import {useTranslation} from 'react-i18next';
+import {useAuth} from '../../../contexts/AuthContext';
+import {Sensor, LabelValue, SensorInput} from './Util';
+
+/**
+ * Props for `DeleteSensorModal`, used for type safety
+ * sensors - all current sensors
+ */
+interface DeleteSensorModalProps {
+  sensors: Sensor[];
+}
+
+/**
+ * Component to delete a sensor. Includes a button to make the modal pop up
+ * and a modal to delete a sensor.
+ */
+const DeleteSensorModal: ({sensors}: DeleteSensorModalProps) => JSX.Element = ({
+  sensors,
+}: DeleteSensorModalProps) => {
+  // --------------- State maintenance variables ------------------------
+  const {isOpen, onOpen, onClose} = useDisclosure();
+  const {isAdmin} = useAuth();
+
+  const [purpleAirId, setPurpleAirId] = useState('');
+  const [sensorDocId, setSensorDocId] = useState('');
+
+  const [confirmPurpleAirId, setConfirmPurpleAirId] = useState('');
+  const [confirmDownload, setConfirmDownload] = useState(false);
+  const [acknowledgeDeletion, setAcknowledgeDeletion] = useState(false);
+  const [acknowledgePermanent, setAcknowledgePermanent] = useState(false);
+
+  // Modal state
+  const [error, setError] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+  // --------------- End state maintenance variables ------------------------
+
+  const readyToSubmit =
+    purpleAirId !== '' &&
+    purpleAirId === confirmPurpleAirId &&
+    error === '' &&
+    confirmDownload &&
+    acknowledgeDeletion &&
+    acknowledgePermanent;
+
+  const {t} = useTranslation(['administration', 'common']);
+
+  /**
+   * Resets modal state values before closing the modal.
+   */
+  function handleClose(): void {
+    // Sensor state
+    setPurpleAirId('');
+    setSensorDocId('');
+
+    // Confirmations
+    setConfirmPurpleAirId('');
+    setConfirmDownload(false);
+    setAcknowledgeDeletion(false);
+    setAcknowledgePermanent(false);
+
+    // Modal state
+    setError('');
+    setIsLoading(false);
+    onClose();
+  }
+
+  /**
+   * This function adds the sensor to the deletion map with the current time,
+   * meaning that all readings in the readings subcollection will be deleted.
+   * If the deletion map is successfully updated, the sensor's doc is deleted in
+   * the sensors collection.
+   * @param event - click button event
+   */
+  function handleDeleteSensor(event: React.MouseEvent) {
+    event.preventDefault();
+
+    if (isAdmin && readyToSubmit) {
+      setIsLoading(true);
+
+      const deletionDocRef = firestore.collection('deletion').doc('todo');
+
+      // Get current mapping or set to empty
+      deletionDocRef
+        .get()
+        .then(deleteDoc => {
+          const newDeletionMap =
+            deleteDoc.data()?.deletionMap ?? Object.create(null);
+
+          newDeletionMap[sensorDocId] = new Date();
+
+          deletionDocRef
+            .update({deletionMap: newDeletionMap})
```

Comment:
did you mention something about using .set instead of .update? Feel free to disregard if that doesn't make sense, just remembering something from someone from this week.

Reply:
Yes, here I think we should use `update` since this doc should already exist and I want to avoid overwriting any data. This does mean that someone cloning this site would need to create this document for this to function properly--perhaps during bug fix/documentation time, we can write a script that would create all necessary collections and docs.